### PR TITLE
TST: Fix 32-bit job, minor workflow updates

### LINKED ISSUE #13112: TST: 32-bit and parallel job failed with setuptools-scm was unable to detect version
This popped up after I merged an one-liner doc PR that should not have affected CI, so I am not sure what triggered it. I reran the job once but it still failed but could still be transient but just using bad cache... who knows. Need to keep an eye on it.

If this won't go away, is it time to stop testing for 32-bit? If so, where are we moving the parallel job to?

Example log: https://github.com/astropy/astropy/runs/6069066705?check_suite_focus=true

```
Traceback (most recent call last):
  File ".tox/.tox/lib/python3.8/site-packages/tox/helper/build_requires.py", line 16, in <module>
    for_build_requires = backend.get_requires_for_build_sdist(None)
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/build_meta.py", line 181, in get_requires_for_build_sdist
    return self._get_build_requires(config_settings, requirements=[])
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/build_meta.py", line [15](https://github.com/astropy/astropy/runs/6069066705?check_suite_focus=true#step:6:15)9, in _get_build_requires
    self.run_setup()
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/build_meta.py", line [17](https://github.com/astropy/astropy/runs/6069066705?check_suite_focus=true#step:6:17)4, in run_setup
    exec(compile(code, __file__, 'exec'), locals())
  File "setup.py", line 68, in <module>
    setup(ext_modules=get_extensions())
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 109, in setup
    _setup_distribution = dist = klass(attrs)
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/dist.py", line 462, in __init__
    _Distribution.__init__(
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line [29](https://github.com/astropy/astropy/runs/6069066705?check_suite_focus=true#step:6:29)3, in __init__
    self.finalize_options()
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools/dist.py", line 886, in finalize_options
    ep(self)
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools_scm/integration.py", line 104, in infer_version
    _assign_version(dist, config)
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools_scm/integration.py", line 51, in _assign_version
    _version_missing(config)
  File "/__w/astropy/astropy/.tox/.package/lib/python3.8/site-packages/setuptools_scm/__init__.py", line 106, in _version_missing
    raise LookupError(
LookupError: setuptools-scm was unable to detect version for /__w/astropy/astropy.
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to close #13112 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by saimn** [comment]
Fine by me since Numpy already dropped i386 and we were pinning it to an older version (#12615).
I guess the main question is what to do for 5.0.x ? Is the job also failing there ?

**Comment by pllim** [comment]
> what to do for 5.0.x ? Is the job also failing there ?

Yes, see https://github.com/astropy/astropy/runs/6087563832?check_suite_focus=true

**Comment by astrofrog** [comment]
Is this due to a new version of setuptools-scm? If so we could potentially pin it in 5.0.x in the pyproject.toml?

**Comment by pllim** [comment]
I didn't see any setuptools-scm version bump between successful job and failed job.

Last Successful: https://github.com/astropy/astropy/runs/6012525149?check_suite_focus=true

First Failed: https://github.com/astropy/astropy/runs/6069066705?check_suite_focus=true

**Comment by pllim** [comment]
setuptools-scm has not had new release since Jan 2022: https://pypi.org/project/setuptools-scm/#history

**Comment by pllim** [comment]
Although https://quay.io/repository/pypa/manylinux2014_i686?tab=tags was tagged a few days ago... but how do we know what changed?

**Comment by astrofrog** [comment]
The docker image was updated 2 days ago - maybe related? If so we could try the ``2022-04-03-da6ecb3`` tag instead of ``latest`` (see https://quay.io/repository/pypa/manylinux2014_i686?tab=tags&tag=latest for a list of tags)

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -143,39 +146,41 @@ jobs:
...
       run: tox ${{ matrix.toxargs }} -e ${{ matrix.toxenv }} -- ${{ matrix.toxposargs }}
 
-  parallel_and_32bit:
-    name: 32-bit and parallel
+  parallel-double:
+    name: Parallel and double
     runs-on: ubuntu-latest
     needs: initial_checks
-    container:
-      image: quay.io/pypa/manylinux2014_i686
```
Comment (astrofrog) [inline .github/workflows/ci_workflows.yml:155]: Here try adding ``:2022-04-03-da6ecb3`` at the end

File: .github/workflows/ci_workflows.yml [hunk 2]
```diff
@@ -143,39 +146,41 @@ jobs:
...
     - name: Install Python dependencies
-      run: python -m pip install --upgrade tox codecov
+      run: python -m pip install --upgrade tox
     - name: Run tests
       run: tox ${{ matrix.toxargs }} -e ${{ matrix.toxenv }} -- ${{ matrix.toxposargs }}
 
-  parallel_and_32bit:
-    name: 32-bit and parallel
+  parallel-double:
+    name: Parallel and double
```
Comment (dhomeier) [inline .github/workflows/ci_workflows.yml:154]: `double` refers to the repeated run? Maybe something with "repeat" or "run twice" would make this clearer; first thing I was thinking of was something with double precision (but perhaps only in memory of the 32-bit job ;-)
Having the Python version stated there would also give a better overview.
And while at that I was wondering if this job should stay at the now oldest supported Python version.

**Review by dhomeier — APPROVED** [review]
Removing 32-bit is OK with me, although I cannot really speak for its relevance at this point.
Only semantic comments otherwise.

**Comment by dhomeier** [comment]
The `_assign_version` stuff was added with setuptools_scm 6.3 and 6.4, but other jobs are also running 6.4.2.
Maybe an existing problem with identifying the version that just went unnoticed before?

File: .github/workflows/ci_workflows.yml (hunk 1 above)
Comment (pllim) [reply .github/workflows/ci_workflows.yml:155]: OK let's see...

File: .github/workflows/ci_workflows.yml (hunk 2 above)
Comment (pllim) [reply .github/workflows/ci_workflows.yml]: `double` naming is irrelevant now if we end up keeping 32-bit, but I'll keep this comment in mind if we change our mind again. Thanks!

I think it is ok to stick with oldest Python version here. We already have to pin to older numpy for reasons.

Comment (dhomeier) [reply .github/workflows/ci_workflows.yml]: I thought the numpy version was also only pinned for the 32-bit env, but if you've managed to save that now, all the better.

Comment (pllim) [reply .github/workflows/ci_workflows.yml]: Right, I unpinned numpy too when I removed 32-bit but now it is back, so this is moot lol.

**Comment by pllim** [comment]
OK 32-bit is green with older Docker image, so they must have bumped setuptools_scm in the newer image. Not sure what we are going to do when they remove that old image.

**Comment by astrofrog** [comment]
> Not sure what we are going to do when they remove that old image.

I don't think that will happen anytime soon. We should probably decide at some point soon about dropping 32-bit support anyway.

**Comment by pllim** [comment]
Okay, CI is green. Shall we merge?

**Comment by dhomeier** [comment]
Thanks for the fix and bringing everything up to snuff!

**Comment by pllim** [comment]
@meeseeksdev backport to v5.0.x

**Pull request merged**